miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/BoxVisualization.java

```diff
@@ -0,0 +1,143 @@
+    WebElement target = getPosition(getRowLabel(position), getColLabel(position));
+    return target.getAttribute("title").contains("Empty");
+  }
+
+  public String getPositionTitle(String position) {
+    WebElement target = getPosition(getRowLabel(position), getColLabel(position));
+    return target.getAttribute("title");
+  }
+
+  protected String getRowLabel(String position) {
+    return position.substring(0, 1);
+  }
+
+  protected int getColLabel(String position) {
+    return Integer.parseInt(position.substring(1));
+  }
+
+  protected WebElement getPosition(String rowLabel, int column) {
+    List<List<WebElement>> trs = visRows.stream().map(row -> row.findElements(tdSelector)).collect(Collectors.toList());
+    if (rowLabel == null || isStringEmptyOrNull(rowLabel)) throw new IllegalArgumentException("Row selector cannot be empty");
+    if (colHeaders.size() < column)
+      throw new IllegalArgumentException("Column selector " + column + " is larger than table size (" + colHeaders.size() + " columns)");
+    List<List<WebElement>> chosenRow = trs.stream()
+        .filter(tdList -> tdList.size() > 0 ? tdList.get(0).getText().trim().equals(rowLabel) : false)
+        .collect(Collectors.toList());
+    if (chosenRow.size() == 0) throw new IllegalArgumentException("Cannot find row " + rowLabel);
+    return chosenRow.get(0).get(column);
+  }
+
+  public void lookupBarcode(String barcode) {
+    barcodeInput.click();
+    barcodeInput.clear();
+    barcodeInput.sendKeys(barcode);
+    barcodeInput.sendKeys(Keys.ESCAPE);
+    barcodeLookupButton.click();
+    waitExplicitly(1000);
+    if (!getDriver().findElements(By.id("ok")).isEmpty()) {
+      WebElement okButton = getDriver().findElement(By.id("ok"));
+      okButton.click();
+    }
+    waitUntil(elementToBeClickable(barcodeLookupButton));
+  }
+
+  public void updatePosition(boolean requireConfirmation) {
+    if (updatePositionButton.getAttribute("disabled") == null) {
+      // click the button if it's clickable, otherwise do nothing
+      updatePositionButton.click();
+      if (requireConfirmation) {
+        WebElement okButton = getDriver().findElement(By.id("ok"));
+        okButton.click();
+        waitUntil(invisibilityOf(okButton));
+      }
+      waitUntil(elementToBeClickable(updatePositionButton));
+    }
+  }
+
+  public void removeTube() {
+    if (removeTubeButton.getAttribute("disabled") == null) {
+      removeTubeButton.click();
+    } else {
+      log.error("* * * * * * * * * Error: removeTubeButton is not clickable * * * * * * * * * ");
+    }
+    confirmAndWaitForRefresh();
+  }
+
+  public void discardTube() {
+    if (discardTubeButton.getAttribute("disabled") == null) {
+      discardTubeButton.click();
+    } else {
+      log.error("* * * * * * * * * Error: discardTubeButton is not clickable * * * * * * * * * ");
+    }
+    confirmAndWaitForRefresh();
+  }
+
+  protected void confirmAndWaitForRefresh() {
+    waitExplicitly(1000);
```

Comment:
`confirmDialog` should wait for presence or visibility of "#ok" instead of explicit wait here